src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -18,94 +17,171 @@
+        try {
+            String singularKey = settings.get(API_KEY);
+            String singularSecret = settings.get(API_SECRET);
+
+            // Getting the DDL timeout from the settings. If does not exist, use 60(S) as default.
+            String ddlTimeout = settings.get(DDL_TIMEOUT);
+            long ddlHandlerTimeoutSec = 60L;
+            if (!KitUtils.isEmpty(ddlTimeout)) {
+                try {
+                    ddlHandlerTimeoutSec = Long.parseLong(ddlTimeout);
+                } catch (Exception unableToGetDDLTimeout) {
+                }
+            }
+
+            SingularConfig config = new SingularConfig(singularKey, singularSecret);
+            config.withDDLTimeoutInSec(ddlHandlerTimeoutSec);
+
+            Activity activity = getCurrentActivity().get();
+
+            if (activity != null) {
+                Intent intent = activity.getIntent();
+
+                config.withSingularLink(intent, new SingularLinkHandler() {
+                    @Override
+                    public void onResolved(SingularLinkParams singularLinkParams) {
+                        AttributionResult attributionResult = new AttributionResult();
+                        attributionResult.setServiceProviderId(MParticle.ServiceProviders.SINGULAR);
+                        attributionResult.setLink(singularLinkParams.getDeeplink());
+                        try {
+                            JSONObject linkParams = new JSONObject();
+                            linkParams.put(PASSTHROUGH, singularLinkParams.getPassthrough());
+                            linkParams.put(IS_DEFERRED, singularLinkParams.isDeferred());
+                            attributionResult.setParameters(linkParams);
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+                        getKitManager().onResult(attributionResult);
+                    }
+                });
+            }
+
+            // If the environment is in development mode, enable logging.
+            if (MPUtility.isDevEnv()) {
+                config.withLoggingEnabled();
+                config.withLogLevel(Log.DEBUG);
+            }
+
+            Singular.setWrapperNameAndVersion(MPARTICLE_WRAPPER_NAME, MPARTICLE_WRAPPER_VERSION);
+
+            return config;
+
+        } catch (Exception ex) {
+            logger.error("Can't build Singular Config in the mParticle Kit", ex);
+            return null;
+        }
+    }
+
     @Override
     public List<ReportingMessage> setOptOut(boolean b) {
         return null;
     }
 
     @Override
     public String getName() {
-        return "Singular";
+        return KIT_NAME;
     }
 
     @Override
-    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
-        return null;
+    public void setInstallReferrer(Intent intent) {
+        new SingularInstallReceiver().onReceive(getContext(), intent);
```

Comment:
Wouldn't it be nice (haha) if there was exactly one of those, just in case someone screws up and it is called multiple times?